var/spack/repos/builtin/packages/pythia8/package.py

```diff
@@ -71,6 +72,16 @@ def configure_args(self):
 
         return args
 
+    def url_for_version(self, version):
+        url, dirname, fname = self.url.rsplit('/', 2)
+        dirname = 'pythia' + str(version)[:2]
+
+        return url + '/' + dirname + '/' + fname
+
     def setup_dependent_run_environment(self, env, dependent_spec):
         env.set('PYTHIA8', self.prefix)
         env.set('PYTHIA8DATA', self.prefix.share.Pythia8.xmldoc)
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('PYTHIA8', self.prefix)
+        env.set('PYTHIA8DATA', self.prefix.share.Pythia8.xmldoc)
```

Comment:
Just an idea:

Could we either have a `setup_common_env` to not repeat all this setup code?

Or some fancy `setup_dependent_build_environment = setup_dependent_run_environment` (they have the same signature, so this should work.

Reply:
Sure.